app/test/specs/postDirective.spec.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+describe('Test PostDirective', function() {
+    beforeEach(module('app'));
+
+    var postCtrl, post, httpBackend, scope, deffered, mdDialog, rootScope, postService, mdToast;
+    var user = {
+        name: 'name',
+        current_institution: {key: "institutuion_key"}
+    };
+   
+    beforeEach(inject(function($controller, $httpBackend, $q, $mdDialog, 
+            PostService, AuthService, $mdToast, $rootScope) {
+        postCtrl = $controller('PostController');
+        httpBackend = $httpBackend;
+        rootScope = $rootScope;
+        scope = $rootScope.$new();
+        deffered = $q.defer();
+        mdDialog = $mdDialog;
+        postService = PostService;
+        mdToast = $mdToast;
+        postCtrl.user = user;
+        post = {
+            title: 'title',
+            text: 'text',
+            institution: {}
+        };
+        httpBackend.when('GET', '/api/user').respond(user);
```

Comment:
usa expect e não when, com o expect você força o teste a olhar para a requisição, caso não ocorra exatamente o número de requests esperados ocorre erro.